internal/spoke_k8s_client/spoke_k8s_client_manager.go

```diff
@@ -0,0 +1,44 @@
+package spoke_k8s_client
+
+import (
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// SpokeK8sClientManager maintains a set of spoke clients
+
+//go:generate mockgen --build_flags=--mod=mod -package=spoke_k8s_client -destination=mock_spoke_k8s_client_manager.go . SpokeK8sClientManager
+type SpokeK8sClientManager interface {
```

Comment:
I would prefer we call this a "Cache" rather than a "Manager".

Reply:
ok, makes sense.